scripts/reinforcement_learning/ray/task_runner.py

```diff
@@ -0,0 +1,176 @@
+import yaml
+import ray
+import sys
+import argparse
+import subprocess
+import threading
+from enum import Enum
+
+"""
+This script dispatches one or more user-defined Python tasks to workers in a Ray cluster.
+Each task, with its resource requirements and execution parameters, is described in a YAML configuration file.
+You may specify the desired number of CPUs, GPUs, and memory allocation for each task in the config file.
+
+Key features:
+- Flexible resource management per task via config fields (`num_gpus`, `num_cpus`, `memory`).
+- Real-time output streaming (stdout/stderr) for each task.
+- Parallel execution of multiple tasks across cluster resources.
+
+Tasks are distributed and scheduled according to Ray’s built-in resource manager.
+
+Typical usage:
+---------------
+
+.. code-block:: bash
+
+    # Print help and argument details:
+    python task_runner.py -h
+
+    # Submit tasks defined in a YAML file to the Ray cluster (auto-detects Ray head address):
+    python task_runner.py --task_cfg /path/to/tasks.yaml
+
+YAML configuration example:
+---------------------------
+.. code-block:: yaml
```

Comment:
Nit: can you show a real example in a config file as well if possible? Maybe will make this example a bit more clear